external_api/api.py

```diff
@@ -0,0 +1,114 @@
+        response['success'] = True
+
+    except (DataStoreSettingsImproperlyConfiguredException,
+            EngineNotImplementedException,
+            EngineConnectionException, FetchIndexForAliasException) as error_message:
+        response['error'] = str(error_message)
+        ner_logger.exception('Error: %s' % error_message)
+        return HttpResponse(json.dumps(response), content_type='application/json', status=500)
+
+    except BaseException as e:
+        response['error'] = str(e)
+        ner_logger.exception('Error: %s' % e)
+        return HttpResponse(json.dumps(response), content_type='application/json', status=500)
+
+    return HttpResponse(json.dumps(response), content_type='application/json', status=200)
+
+
+def update_dictionary(request):
+    """
+    This function is used to update the dictionary entities.
+    Args:
+        request (HttpResponse): HTTP response from url
+
+    Returns:
+        HttpResponse : HttpResponse with appropriate status and error message.
+    """
+    response = {"success": False, "error": ""}
+    try:
+        external_api_data = json.loads(request.GET.get(EXTERNAL_API_DATA))
+        dictionary_name = external_api_data.get(DICTIONARY_NAME)
+        dictionary_data = external_api_data.get(DICTIONARY_DATA)
+        language_script = external_api_data.get(LANGUAGE_SCRIPT)
+        datastore_obj = DataStore()
+        datastore_obj.external_api_update_entity(dictionary_name=dictionary_name,
+                                                 dictionary_data=dictionary_data,
+                                                 language_script=language_script)
+        response['success'] = True
+
+    except (DataStoreSettingsImproperlyConfiguredException,
+            EngineNotImplementedException,
+            EngineConnectionException, FetchIndexForAliasException) as error_message:
+        response['error'] = str(error_message)
+        ner_logger.exception('Error: %s' % error_message)
+        return HttpResponse(json.dumps(response), content_type='application/json', status=500)
+
+    except BaseException as e:
+        response['error'] = str(e)
+        ner_logger.exception('Error: %s' % e)
+        return HttpResponse(json.dumps(response), content_type='application/json', status=500)
+    return HttpResponse(json.dumps(response), content_type='application/json', status=200)
+
+
+def transfer_entities(request):
+    """
+    This method is used to transfer entities from the source to destination.
+    Args:
+        request (HttpResponse): HTTP response from url
+    Returns:
+        HttpResponse : HttpResponse with appropriate status and error message.
+    """
+    response = {"success": False, "error": ""}
+    try:
+        external_api_data = json.loads(request.GET.get(EXTERNAL_API_DATA))
+        entity_list = external_api_data.get(ENTITY_LIST)
+
+        datastore_object = DataStore()
+        datastore_object.transfer_entities(entity_list=entity_list)
```

Comment:
Refer to comments in traansfer_entities definition

Reply:
Done